app/gui2/src/stores/graph/graphDatabase.ts

```diff
@@ -517,7 +521,9 @@ export interface NodeDataFromMetadata {
   vis: Opt<VisualizationMetadata>
 }
 
-export interface Node extends NodeDataFromAst, NodeDataFromMetadata {}
+export interface Node extends NodeDataFromAst, NodeDataFromMetadata {
+  zIndex?: number
```

Comment:
I assume that right now the default state here is `undefined` (or rather, as missing field). I think it will cause less issues down the line if we make this field required and explicitly set the `zIndex` for each created node. That newly nodes will always be rendered on top, instead of being under all nodes that happened to have `zIndex` set to some value.